torch/csrc/autograd/python_function.cpp

```diff
@@ -744,7 +744,9 @@ PyObject * THPFunction_do_backward(THPFunction *self, PyObject *args)
       THPUtils_invalidArguments(args, nullptr, "_do_backward", 1, "(tuple, bool)");
       return nullptr;
     }
-    auto cdata = self->cdata.lock();
+
+    auto self_thp = (THPFunction*)self;
```

Comment:
nit: you can use the same `_self` trick as above here to keep `self` in the rest of the function?
There are other uses of `self` but they should work with either pointer I think.